apps/genpkey.c

```diff
@@ -294,14 +322,21 @@ int genpkey_main(int argc, char **argv)
 
  end:
     sk_OPENSSL_STRING_free(keyopt);
-    if (ret != 0)
+    if (ret != 0) {
         ERR_print_errors(bio_err);
+    } else {
+        rv = write_to_file(mem_outpubkey, outpubkeyfile, outformat, private);
```

Comment:
This needs to be guarded by `mem_outputkey != NULL` and skipped otherwise.

Reply:
Updated